cmd/up.go

```diff
@@ -142,20 +152,45 @@ func isGKE(disco discovery.DiscoveryInterface) (bool, error) {
 	return false, nil
 }
 
-func generateMongoDBSecret(cli kubernetes.Interface, objs []*unstructured.Unstructured) error {
-	var mongoNS string
-	for _, obj := range objs {
-		if obj.GetKind() == "Deployment" && obj.GetName() == "mongodb" {
-			mongoNS = obj.GetNamespace()
-			break
-		}
-	}
-	if mongoNS == "" {
-		return fmt.Errorf("Unable to find the namespace of the MongoDB database")
+func mongoSecretExists(c kubecfg.ApplyCmd) (*unstructured.Unstructured, bool, error) {
+	gvk := schema.GroupVersionKind{Version: "v1", Kind: "Secret"}
+	rc, err := clientForGroupVersionKind(c.ClientPool, c.Discovery, gvk, Kubeapps_NS)
+	if err != nil {
+		return nil, false, err
 	}
-	err := populateSecretWithPasswords(cli, mongoNS, "mongodb", []string{"mongodb-password", "mongodb-root-password"})
+	pwFields := []string{"mongodb-password", "mongodb-root-password"}
```

Comment:
if you leave these `pwFields` as a parameter the function becomes generic (which is better)